### PR TITLE
#417 - Added SecureSlice and AuthSecureSlice

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #417 
Added `SecureSlice` and `AuthSecureSlice`

### CODE REVIEW

**Review by olenagerasimova — APPROVED** [review]
@olegmoz looks good to merge

File: src/main/java/com/artipie/docker/http/SecureSlice.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package com.artipie.docker.http;
+
+import com.artipie.http.Slice;
+
+/**
+ * Slice requiring authorization specified by {@link Scope}.
+ *
+ * @since 0.11
+ */
+public interface SecureSlice extends Slice {
```
Comment (genryxy) [inline src/main/java/com/artipie/docker/http/SecureSlice.java:33]: @olegmoz probably to rename this interface to `ScopeSlice` or `SecureScopeSlice`

Comment (olegmoz) [reply src/main/java/com/artipie/docker/http/SecureSlice.java]: @genryxy thanks, changed the names

**Review by genryxy — APPROVED** [review]
@olegmoz thanks, approve

**Pull request merged**